core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -700,7 +700,7 @@ public void processPendingDeletes(Index index, IndexSettings indexSettings, Time
                             sleepTime = Math.min(maxSleepTimeMs, sleepTime * 2); // increase the sleep time gradually
                             logger.debug("{} schedule pending delete retry after {} ms", index, sleepTime);
                         } catch (InterruptedException e) {
-                            Thread.interrupted();
+                            Thread.currentThread().interrupt();
```

Comment:
This one is a bit more tricky, but there are also other calls that also set the interruption flag already (see `InternalShardLock#acquire()` which is called by `NodeEnvironment#lockAllForIndex()`). After looking what happens in the error handling code for this case, I'd opt for rethrowing `InterruptedException` directly instead of just setting the flag. In conjunction with that this shows that it makes sense that `AbstractRunnable` restores the flag on `InterruptedException` then (this whole code block is enclosed in a subclass of `AbstractRunnable`).